lib/filewatch/watch.rb

```diff
@@ -62,62 +45,66 @@ def initialize(opts={})
       end
       @watching = []
       @exclude = []
-      @files = Hash.new { |h, k| h[k] = WatchedFile.new(k, false, false) }
-      @unwatched = Hash.new
+      @files = Hash.new
       # we need to be threadsafe about the mutation
       # of the above 2 ivars because the public
-      # methods each, discover, watch and unwatch
+      # methods each, discover and watch
       # can be called from different threads.
       @lock = Mutex.new
       # we need to be threadsafe about the quit mutation
       @quit = false
       @quit_lock = Mutex.new
+      @max_active = ENV.fetch("FILEWATCH_MAX_OPEN_FILES", 4095).to_i
+      @max_warn_msg = "Reached maximum (#{@max_active}) open files"
+      @lastwarn_max_files = 0
     end # def initialize
 
     public
+
+    def max_open_files=(value)
+      val = value.to_i
+      return if value.nil? || val <= 0
+      @max_warn_msg = "Reached maximum (#{val}) open files"
```

Comment:
I believe Jordan's comment was more oriented towards communication with the user, not just developer. On that I agree that the "Reached maximum .. open files" should be clearer that it's a filewatch threshold, not OS.

Maybe "Reached \'max_open_files\' threshold (#{ ..})." ? I'm not a fan of this either..ideas?